diff --git a/toolchain/BUILD.llvm_repo.tpl b/toolchain/BUILD.llvm_repo.tpl
index ab27cdf..2fe1545 100644
--- a/toolchain/BUILD.llvm_repo.tpl
+++ b/toolchain/BUILD.llvm_repo.tpl
@@ -61,6 +61,12 @@ filegroup(
     ),
 )
 
+cc_library(
+    name = "clang_headers",
+    hdrs = glob(["include/**"]),
+    strip_include_prefix = "include",
+)
+
 # This filegroup should only have source directories, not individual files.
 # We rely on this assumption in system_module_map.bzl.
 filegroup(
