--- a/compiler/plc_driver/src/pipelines.rs
+++ b/compiler/plc_driver/src/pipelines.rs
@@ -424,6 +424,11 @@
             return Ok(());
         }
 
+        if self.compile_parameters.as_ref().is_some_and(|opt| opt.print_ast_json) {
+            println!("{}", serde_json::to_string_pretty(&parsed_project.units).expect("AST is always serializable"));
+            return Ok(());
+        }
+
         // 2. Index
         let indexed_project = self.index(parsed_project)?;
 
